x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/RetentionPolicyToDeleteByQueryRequestConverter.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.transform.transforms;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.reindex.AbstractBulkByScrollRequest;
+import org.elasticsearch.index.reindex.DeleteByQueryRequest;
+import org.elasticsearch.xpack.core.transform.transforms.DestConfig;
+import org.elasticsearch.xpack.core.transform.transforms.RetentionPolicyConfig;
+import org.elasticsearch.xpack.core.transform.transforms.SettingsConfig;
+import org.elasticsearch.xpack.core.transform.transforms.TimeRetentionPolicyConfig;
+import org.elasticsearch.xpack.core.transform.transforms.TransformCheckpoint;
+
+import java.time.Instant;
+
+/**
+ * Implementation of `retention_policy` configuration parameter.
+ *
+ * All implementations of `rentention_policy` are converted to a {@link DeleteByQueryRequest}, which is than executed by the indexer.
+ */
+public final class RetentionPolicyToDeleteByQueryRequestConverter {
+
+    public static class RetentionPolicyException extends ElasticsearchException {
+        RetentionPolicyException(String msg, Object... args) {
+            super(msg, args);
+        }
+    }
+
+    private RetentionPolicyToDeleteByQueryRequestConverter() {}
+
+    /**
+     * Build a {@link DeleteByQueryRequest} from a retention policy. The DBQ should run _after_ a new checkpoint has finished.
+     * The given checkpoint should be the one that just finished indexing, however the DBQ executes before this checkpoint
+     * gets exposed.
```

Comment:
Could you clarify how is the checkpoint made "exposed" other than just being "indexed"?